jsdom-nodejs-env/src/test/scala/org/scalajs/jsenv/jsdomnodejs/TestKit.scala

```diff
@@ -0,0 +1,159 @@
+package org.scalajs.jsenv.jsdomnodejs
+
+// !!! This entire file is copied from TestKit.scala upstream
```

Comment:
Do you really need all this? From what I can see is all you want to test is that the window API doesn't fail (I guess that is to check there is a DOM).

Wouldn't it be enough to check:
- That the code `window.history.pushState({}, "", "/foo"); window.close()` terminates successfully.
- That the code `window.history.inexistent();` does fail.

That way you only need to rely on the public interface of the Env.

Reply:
Well `TestKit` only relies on the public API of `JSEnv` too. I first tried to do it by hand, but quickly ended up redoing several steps already provided by the test kit: handling of futures, comparison of the output, etc. Eventually I decided that I would have less to maintain by copying it over.

Reply:
> Well TestKit only relies on the public API of JSEnv too.

Yes, but your test will then also rely on `TestKit`.

I understand that output handling is difficult, but do you need to read output here? Future handling would be just calling `Await.result`, or am I missing something?

Reply:
I added a commit that shows what the test looks like if I don't use `TestKit`. It does not read the output, which indeed makes the thing simpler. It would be significantly more complicated if it did have to check the output, though.

Should I proceed with the new test, and rip out the one using `TestKit`?

Reply:
Yes, let's go ahead with this for now and then re-asses whether we need to expose something `TestKit` like (IMHO `TestKit` needs better design before we expose it).